File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")

const n = Number(input[0])
let answer =0

let nicknames = new Set()

for(let str of input.slice(1)){
  if(str === 'ENTER'){
    answer += nicknames.size
    nicknames.clear()
  }else{
    nicknames.add(str)
  }
}
answer += nicknames.size
console.log(answer)

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys

input = sys.stdin.readline

n = int(input())
answer =0
nicknames = set()

strs = [input().strip() for i in range(n)]

for i in strs:
    if(i == 'ENTER'):
        answer += len(nicknames)
        nicknames.clear()
    else:
        nicknames.add(i)

answer +=len(nicknames)
print(answer)